### DISCUSSION TITLE
Duration Picker

### DISCUSSION THREAD

**Discussion by kajal-g-7span** [directus/directus#22130] [Feature Requests]
### Summary

Currently, within Directus, while we have a time picker available under the datetime datatype, we lack a specific duration picker to facilitate the storage of durations.

### Motivation

Currently, we encounter difficulty storing durations in Directus. The platform offers the time datatype under Datetime, which only supports a 24-hour format. Consequently, storing durations exceeding 24 hours poses a challenge.

### Detailed Design

<img width="602" alt="image" src="https://github.com/directus/directus/assets/114467240/57de0500-7687-4232-a0ee-683ecfdd3add">

Currently, although we have the aforementioned field available for storing time, it's restricted to durations of up to 24 hours.

<img width="818" alt="image" src="https://github.com/directus/directus/assets/114467240/14e47616-261e-40d3-bed6-6a51f54ddfc7">

We're seeking support for a Duration Picker feature that would allow us to store duration values beyond this limitation.

### Must Have:
- Having support for storing the Duration of time in the format of `HH:MM:SS`

### Should Have:
- Should have 3 fields to store the hours, minutes, and seconds values as Integer value

### Could Have:
- Auto tab feature to do movement between the fields the fields
 
### Won't Have:
- `Set to Now` button

### Drawbacks

We can make custom changes in the component to provide support for the Duration picker.

### Alternatives

Users have to do custom modifications to provide support for Duration Datatype.

### Adoption Strategy

This change is not a breaking change. We have to update our existing component to provide support for Duration.

**Comment by emilyf** [comment]
I would see this as most helpful in storing things like video footage, in which case support for frames would be a good addition (hh:mm:ss:ff). There are definite limitations to it being a datetime field for storage of length of a video, especially if there is a transcode operation that will then be associated with it or an automation with another system (such as a playback system) where frames may be required and where this being "exact" can be crucial for such systems.

**Comment by gla23** [comment]
Duration pickers often store a duration in [ISO 8601](https://www.digi.com/resources/documentation/digidocs/90001488-13/reference/r_iso_8601_duration_format.htm) in order to easily store longer lengths of time. You don't want to manually convert years and months to hours and days, especially when different months and years have different lengths anyway!

The ISO 8601 format also means that it makes sense to use a text inputs to create this string, which is a much nicer UI anyway. See this example from Sentry where I'm searching for `2` and it gives autocompletion options for different lengths.

<img width="257" alt="Screenshot 2024-07-29 at 14 58 00" src="https://github.com/user-attachments/assets/7c5947b8-1385-49f5-bb84-c0e650bcd28a">
